pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java

```diff
@@ -766,39 +768,37 @@ protected void resumeAsyncResponseExceptionally(AsyncResponse asyncResponse, Thr
     }
 
     protected CompletableFuture<SchemaCompatibilityStrategy> getSchemaCompatibilityStrategyAsync() {
-        return validateTopicOperationAsync(topicName, TopicOperation.GET_SCHEMA_COMPATIBILITY_STRATEGY)
-                .thenCompose((__) -> {
-                    CompletableFuture<SchemaCompatibilityStrategy> future;
-                    if (config().isTopicLevelPoliciesEnabled()) {
-                        future = getTopicPoliciesAsyncWithRetry(topicName)
-                                .thenApply(op -> op.map(TopicPolicies::getSchemaCompatibilityStrategy).orElse(null));
-                    } else {
-                        future = CompletableFuture.completedFuture(null);
-                    }
+        validateTopicPolicyOperation(topicName, PolicyName.SCHEMA_COMPATIBILITY_STRATEGY, PolicyOperation.READ);
```

Comment:
This is blocking method. Please use `validateTopicPolicyOperationAsync`.

Reply:
I found that pulsar does not provide async method `validateTopicPolicyOperationAsync`, maybe we should add this method and replace all synchronous method `validateTopicPolicyOperation`.

Reply:
https://github.com/apache/pulsar/blob/a0fe8562694013b11910b39ebf4dc703c1400bf4/pulsar-broker/src/main/java/org/apache/pulsar/broker/web/PulsarWebResource.java#L1101 
This one?